pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -84,18 +85,31 @@ func Add(ctx context.Context, cfg *rest.Config, namespace string) error {
 		return errors.Wrap(err, "could not create metrics Service")
 	}
 
-	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
-	// necessary to configure Prometheus to scrape metrics from this operator.
-	services := []*v1.Service{service}
-	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+	// Create a monitoring client to hold the ServiceMonitor object
+	mclient, err := monclient.NewForConfig(cfg)
 	if err != nil {
-		log.Error(err, "could not create ServiceMonitor object")
-		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
-		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
-		if err == metrics.ErrServiceMonitorNotPresent {
-			metricsEnabled = false
-			return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")
+		return errors.Wrap(err, "could not create monitoring client")
+	}
 
+	// In the case of an operator restart, a previous SM object will be deleted and a
+	// new one will be created with the current spec. If a SM object does not yet exist,
+	// one will be created.
+	deleteErr := mclient.ServiceMonitors(namespace).Delete(context.TODO(), windowsMetricsEndpoints, metav1.DeleteOptions{})
```

Comment:
Instead of calling `createServiceMonitor` twice and creating a separate function, you could check for error at the start
```suggestion

	deleteErr := mclient.ServiceMonitors(namespace).Delete(context.TODO(), windowsMetricsEndpoints, metav1.DeleteOptions{})
	if deleteErr != nil  && !apierrors.IsNotFound(err) {
		return errors.Wrap(err, "could not delete existing ServiceMonitor object")
	}  
	
	
	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
	// necessary to configure Prometheus to scrape metrics from this operator.
	services := []*v1.Service{service}
	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
	if err != nil {
		log.Error(err, "could not create ServiceMonitor object")
		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
		if err == metrics.ErrServiceMonitorNotPresent {
			metricsEnabled = false
			return errors.Wrap(err, "install prometheus-operator in your cluster to create ServiceMonitor objects")

		}
	}
```

Reply:
Ok thanks! This looks like it will work if the operator is restarting but what about when it is the first time the operator is being deployed? A SM object won't yet exist so deleteErr will be != nil, will it just jump to SM creation?

Reply:
It should, since we are moving to `creation`  if Service Monitor object is deleted successfully or is not found

Reply:
Ok testing it now

Reply:
I tested this approach but it fails to create a SM. I ran `oc -n openshift-windows-machine-config-operator get servicemonitor` and it returned `No resources found in openshift-windows-machine-config-operator namespace.`

Reply:
These changes worked! I just added them to the PR